controllers/clustergroupupgrade_controller.go

```diff
@@ -171,7 +171,8 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 		}
 	}
 
-	if clusterGroupUpgrade.Status.Backup == nil || meta.IsStatusConditionTrue(clusterGroupUpgrade.Status.Conditions, BackupStateDone) {
```

Comment:
Wouldn't the below condition work as it was before work?

`if clusterGroupUpgrade.Status.Backup == nil || meta.IsStatusConditionTrue(clusterGroupUpgrade.Status.Conditions, utils.ConditionTypes.BackupSuceeded)`

Reply:
It would however in the next PR I will need to check the condition for whether it was partially or fully completed. I guess I put the variable here instead by accident but it will be used more in the follow up